src/components/Home.test.js

```diff
@@ -1,17 +1,78 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-import Home from './Home';
+import { HomeComponent, HomeButtonsMajor, HomeButtonsMinor } from './Home';
+import { Link } from 'react-router-dom';
+Link.displayName = 'Link';
+jest.mock('@material-ui/core/Button', () => 'Button');
 jest.mock('@material-ui/core/Grid', () => 'Grid');
+jest.mock('@material-ui/core/IconButton', () => 'IconButton');
+jest.mock('@material-ui/icons/Call', () => 'CallIcon');
+jest.mock('@material-ui/icons/Restaurant', () => 'RestaurantIcon');
+jest.mock('@material-ui/icons/Home', () => 'HomeIcon');
+jest.mock('@material-ui/icons/HotTub', () => 'HotTubIcon');
+jest.mock('@material-ui/icons/DirectionsBus', () => 'BusIcon');
+jest.mock('@material-ui/icons/LocalGroceryStore', () => 'GroceryStoreIcon');
+jest.mock('@material-ui/icons/Healing', () => 'HealingIcon');
+jest.mock('@material-ui/icons/LibraryBooks', () => 'LibraryBooksIcon');
+jest.mock('@material-ui/icons/Pets', () => 'PetsIcon');
+jest.mock('@material-ui/icons/Wifi', () => 'WifiIcon');
+jest.mock('@material-ui/icons/Work', () => 'WorkIcon');
+jest.mock('@material-ui/icons/Info', () => 'InfoIcon');
 jest.mock('./Search', () => 'Search');
 
 describe('<Home/>', () => {
-  const wrapper = shallow(<Home />);
+  const wrapper = shallow(<HomeComponent />);
+
+  const numSearchGridComponents = 3;
+  const numButtonsMajorGridComponents = 2 + HomeButtonsMajor.length * 2;
+  const numButtonsMinorGridComponents = 2 + HomeButtonsMinor.length * 2;
   it('renders the proper number of layout components', () => {
-    expect(wrapper.find('Grid').length).toBe(2);
+    expect(wrapper.find('Grid').length).toBe(
+      1 +
+        numSearchGridComponents +
+        numButtonsMajorGridComponents +
+        numButtonsMinorGridComponents
+    );
   });
   it('renders one layout component as the search component', () => {
     expect(
       wrapper.findWhere(n => n.prop('component') === 'Search').length
     ).toBe(1);
   });
+  it('renders all major buttons and only once', () => {
+    HomeButtonsMajor.forEach(button => {
+      let matchingButtons = wrapper
+        .findWhere(child => {
+          return (
+            child.type() === 'Button' &&
+            child.text() === button.text &&
+            child.find(`${button.icon}`).length === 1 &&
+            child.prop('component') &&
+            child.prop('component') === Link &&
+            child.prop('to') === button.to
+          );
+        })
+        .map(iconButton => iconButton.text());
+      expect(matchingButtons).toContain(button.text);
+      expect(matchingButtons.length).toBe(1);
+    });
+  });
+  it('renders all minor buttons and only once', () => {
+    HomeButtonsMinor.forEach(button => {
+      let matchingButtons = wrapper
+        .findWhere(child => {
+          return (
+            child.type() === 'Button' &&
```

Comment:
to match the button this returned function must evaluate to true, correct? is checking all of these properties on `child` necessary? I'm not super familiar with `findWhere`, just seems like overkill

Reply:
This is just making sure that the compared element is a Button component and then dives into more specific properties based on if that is a true base.

I could technically do `wrapper.find('Button').findWhere(child => {...})` and accomplish the same thing.

The reason I think it's necessary to test all of these props is that I want to make sure that each button prop has the correct value assigned to it. So I in the props below, I'm testing:

1. It's a <Button/>
2. It contains the correct icon
3. It contains a `component` prop that is a <Link/> component.
4. It contains a `to` prop equal to the path specified.

Maybe each one of these could be their own test?

Reply:
Yeah, i'd pull them out to their own tests. typically we try to keep a single expectation with a single test. I do see your reasoning here though.